app/src/main/java/com/example/parkingsystem/mvp/view/ParkingSpaceReservationView.java

```diff
@@ -63,4 +64,80 @@ public void showTimePickerDialog(TimePickerDialog.OnTimeSetListener listener) {
     public void enableButtonEnd() {
         binding.buttonParkingSpaceReservationPickerEnd.setEnabled(true);
     }
+
+    @Override
+    public int getParkingSpace() {
+        int ParkingSpace = Integer.parseInt(binding.inputParkingSpaceReservationPlace.getText().toString());
+        return ParkingSpace;
+    }
+
+    @Override
+    public int getSecurityCode() {
+        int securityCode = Integer.parseInt(binding.inputParkingSpaceReservationCode.getText().toString());
+        return securityCode;
+    }
+
+    @Override
+    public void backToActivityMain() {
+        Activity activity = getActivity();
+        if (activity != null) {
+            activity.finish();
+        }
+    }
+
+    @Override
+    public void showSaveMessage() {
+        Context context = getContext();
+        if (context != null) {
+            Toast.makeText(context, context.getString(R.string.toast_parking_space_reservation_save), Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    @Override
+    public void showMissingDateStartToast() {
+        Context context = getContext();
+        if (context != null) {
+            Toast.makeText(context, context.getString(R.string.toast_parking_space_reservation_missing_date_start), Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    @Override
+    public void showMissingTimeStartToast() {
+        Context context = getContext();
+        if (context != null) {
+            Toast.makeText(context, context.getString(R.string.toast_parking_space_reservation_missing_time_start), Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    @Override
+    public void showMissingDateEndToast() {
+        Context context = getContext();
+        if (context != null) {
+            Toast.makeText(context, context.getString(R.string.toast_parking_space_reservation_missing_date_end), Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    @Override
+    public void showMissingTimeEndToast() {
+        Context context = getContext();
+        if (context != null) {
+            Toast.makeText(context, context.getString(R.string.toast_parking_space_reservation_missing_time_end), Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    @Override
+    public void showMissingParkingSpaceToast() {
+        Context context = getContext();
+        if (context != null) {
+            Toast.makeText(context, context.getString(R.string.toast_parking_space_reservation_missing_place), Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    @Override
+    public void showMissingSecurityCodeToast() {
+        Context context = getContext();
+        if (context != null) {
+            Toast.makeText(context, context.getString(R.string.toast_parking_space_reservation_missing_security_code), Toast.LENGTH_SHORT).show();
+        }
+    }
```

Comment:
Severals method do the same, show a toast, so try to use some method to include the same part of code,

Reply:
Ok! This is part of the validation of empty fields, so it would be changed in the next PR